File: java/src/main/java/gov/doc/ntia/sigmf/ext/annotation/algorithm/DomainDetection.java
package gov.doc.ntia.sigmf.ext.annotation.algorithm;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import gov.doc.ntia.sigmf.Annotation;

import java.io.Serializable;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class DomainDetection extends Annotation  {

    private static final long serialVersionUID = 1L;

    @JsonProperty(value="ntia-algorithm:detector", required =  true)
    protected String detector;

    @JsonProperty(value="ntia-algorithm:units", required = true)
    protected String units;

    @JsonProperty(value="ntia-algorithm:reference", required = false)
    protected String reference;


    public String getDetector() {
        return detector;
    }

    public void setDetector(String detector) {
        this.detector = detector;
    }

    public String getUnits() {
        return units;
    }

    public void setUnits(String units) {
        this.units = units;
    }

    public String getReference() {
        return reference;
    }

    public void setReference(String reference) {
        this.reference = reference;
    }

}

File: Csharp/sigmf/sigmf/ext/Annotationdir/algorithm/DomainDetection.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf
{
    [Serializable()]
    public class DomainDetection : Annotation
    {
        private static readonly long serialVersionUID = 1L;

        [JsonProperty("ntia-algorithm:detector", Required = Required.Always)]
        protected string detector;

        [JsonProperty("ntia-algorithm:detection_domain", Required = Required.Always)]
        protected string detectionDomain;

        [JsonProperty("ntia-algorithm:units", Required = Required.Always)]
        protected string units;

        [JsonProperty("ntia-algorithm:reference", Required = Required.Always)]
        protected string reference;

        public string Detector { get => detector; set => detector = value; }
        public string DetectionDomain { get => detectionDomain; set => detectionDomain = value; }
        public string Units { get => units; set => units = value; }
        public string Reference { get => reference; set => reference = value; }
    }
}
